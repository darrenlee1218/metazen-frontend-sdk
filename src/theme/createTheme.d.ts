import { Theme as SystemTheme, ThemeOptions } from '@mui/system';
import { CustomTheme, Palette } from '@mui/material/styles';

declare module '@mui/material/styles' {
  export interface ColorsInterface {
    primary: string;
    appBackground: string;
    inputBackground: string;
    bottomNavBackground: string;
    closeButtonBackground: string;
    primaryText: string;
    secondaryText: string;
    placeholderText: string;
    clickableGray: string;
    border: string;
    green: string;
    clickableGrayHover: string;
    hoverColor: string;
    disabledColor: string;
    nonClickableGray: string;
    loadingPrimary: string;
    loadingSecondary: string;
    errorPrimary: string;
    errorSecondary: string;
    successPrimary: string;
    successSecondary: string;
    warningPrimary: string;
    warningSecondary: string;
  }

  interface CustomPalette extends Palette {
    colors: ColorsInterface;
  }

  interface CustomTheme extends SystemTheme {
    palette: CustomPalette;
  }
  // allow configuration using `createTheme`
  interface CustomThemeOptions extends ThemeOptions {
    palette?: CustomPalette;
  }

  export function createTheme(options?: CustomThemeOptions): CustomTheme;
}

declare module '@emotion/react' {
  interface Theme extends CustomTheme {}
}
