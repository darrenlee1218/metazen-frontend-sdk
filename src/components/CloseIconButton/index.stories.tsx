import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import CloseIconButton from '.';

export default {
  title: 'common-components/stateless/atoms/CloseIconButton',
  component: CloseIconButton,
  argTypes: {},
} as ComponentMeta<typeof CloseIconButton>;

const Template: ComponentStory<typeof CloseIconButton> = (args) => <CloseIconButton {...args} />;

export const Default = Template.bind({});

Default.args = {
  onClick: () => {},
};
