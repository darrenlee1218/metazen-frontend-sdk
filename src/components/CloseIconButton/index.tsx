import React from 'react';
import IconButton from '@mui/material/IconButton';
import CloseIcon from '@mui/icons-material/Close';
import { CustomTheme, useTheme } from '@mui/material/styles';

interface CloseIconButtonProps {
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const CloseIconButton: React.FC<CloseIconButtonProps> = ({ onClick }) => {
  const theme = useTheme() as CustomTheme;
  return (
    <IconButton
      onClick={onClick}
      size="small"
      sx={{ backgroundColor: theme.palette.colors.closeButtonBackground, width: '24px', height: '24px' }}
      data-testid="close-icon-button"
    >
      <CloseIcon fontSize="small" sx={{ color: theme.palette.text.primary, fontSize: '16px' }} />
    </IconButton>
  );
};

export default CloseIconButton;
