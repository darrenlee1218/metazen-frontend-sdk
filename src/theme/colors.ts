import { ColorsInterface } from '@mui/material/styles';

const Colors: ColorsInterface = {
  // game/animoca brands primary color(s)
  // background
  appBackground: '#111619',
  inputBackground: '#0d1014',
  bottomNavBackground: 'rgba(55, 71, 80, 0.55)',
  closeButtonBackground: 'rgba(255, 255, 255, 0.15)',
  // Text
  primaryText: '#ffffff',
  secondaryText: '#8A9296',
  placeholderText: '#47505D',
  // others
  clickableGray: '#20282D',
  border: '#20282d',
  green: '#67d65b',
  clickableGrayHover: '#303B43',
  primary: '#12ABEC',
  hoverColor: '#3BBBF1',
  disabledColor: '#04394F',
  nonClickableGray: '#171D21',

  loadingPrimary: '#42CCEC',
  loadingSecondary: '#0A2930',
  errorPrimary: '#FA4D56',
  errorSecondary: '#2C0608',
  successPrimary: '#67D65B',
  successSecondary: '#0E250C',
  warningPrimary: '#EB7C4A',
  warningSecondary: '#20140E',
};

export default Colors;
