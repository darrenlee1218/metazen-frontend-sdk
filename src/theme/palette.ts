import { DeepPartial } from '@gryfyn-types/generics';
import { CustomPalette } from '@mui/material/styles';
import Colors from './colors';
import { Mode } from './types';

export const commonPallete = {
  mode: Mode.Dark,
  primary: {
    main: Colors.primary,
  },
  colors: {
    ...Colors,
  },
  text: {
    primary: Colors.primaryText,
    secondary: Colors.secondaryText,
  },
  background: {
    default: Colors.appBackground,
    paper: Colors.appBackground,
  },
  loading: {
    main: Colors.loadingPrimary,
  },
  error: {
    main: Colors.errorPrimary,
  },
  info: {
    main: '#42CCEC',
  },
  success: {
    main: Colors.successPrimary,
  },
  warning: {
    main: Colors.warningPrimary,
  },
};

export type AdditionalPalette = DeepPartial<CustomPalette>;

const palette = (additionalPalette?: AdditionalPalette) =>
  ({
    ...commonPallete,
    ...additionalPalette,
    colors: {
      ...commonPallete.colors,
      ...additionalPalette?.colors,
    },
  } as unknown as CustomPalette);

export default palette;
