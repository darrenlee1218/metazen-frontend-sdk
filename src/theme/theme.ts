import { createTheme } from '@mui/material/styles';
import { ButtonProps } from '@mui/material/Button';
import palette, { AdditionalPalette } from './palette';
import colors from './colors';

export const defaultTheme = (additionalPalette?: AdditionalPalette) =>
  createTheme({
    palette: palette(additionalPalette),
    typography: {
      fontFamily: ['Basier Circle', 'graphie'].join(','),
      h6: {
        fontSize: '8px',
      },
      h5: {
        fontSize: '10px',
      },
      h4: {
        fontSize: '12px',
      },
      h3: {
        fontSize: '14px',
      },
      h2: {
        fontSize: '16px',
      },
      h1: {
        fontSize: '24px',
      },
    },
    shape: {
      borderRadius: 1,
    },
    breakpoints: {
      values: {
        xs: 0,
        sm: 600,
        md: 900,
        lg: 1200,
        xl: 1536,
      },
    },
    components: {
      MuiCssBaseline: {
        styleOverrides: `
        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 400;
          src: local('Basier Circle Regular'), local('BasierCircle-Regular'), url('./fonts/BasierCircle-Regular.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 400;
          src: local('Basier Circle Regular Italic'), local('BasierCircle-RegularItalic'), url('./fonts/BasierCircle-RegularItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 500;
          src: local('Basier Circle Medium'), local('BasierCircle-Medium'), url('./fonts/BasierCircle-Medium.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 500;
          src: local('Basier Circle Medium Italic'), local('BasierCircle-MediumItalic'), url('./fonts/BasierCircle-MediumItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 600;
          src: local('Basier Circle SemiBold'), local('BasierCircle-SemiBold'), url('./fonts/BasierCircle-SemiBold.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 600;
          src: local('Basier Circle SemiBold Italic'), local('BasierCircle-SemiBoldItalic'), url('./fonts/BasierCircle-SemiBoldItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 700;
          src: local('Basier Circle Bold'), local('BasierCircle-Bold'), url('./fonts/BasierCircle-Bold.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 700;
          src: local('Basier Circle Bold Italic'), local('BasierCircle-BoldItalic'), url('./fonts/BasierCircle-BoldItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }
        input:-webkit-autofill,
        input:-webkit-autofill:focus {
          transition: background-color 600000s 0s, color 600000s 0s;
          box-shadow: none;
        }
      `,
      },
      MuiMobileStepper: {
        styleOverrides: {
          dot: {
            backgroundColor: colors.placeholderText,
          },
          dotActive: {
            backgroundColor: colors.primaryText,
          },
        },
      },
      MuiOutlinedInput: {
        styleOverrides: {
          root: {
            backgroundColor: colors.inputBackground,
            borderRadius: 4,
            input: {
              fontSize: 12,
              '&::placeholder': {
                opacity: 1,
              },
              '&:-webkit-autofill': {
                '-webkitBoxShadow': `0 0 0px 1000px ${colors.inputBackground} inset`,
                '-webkit-text-fill-color': colors.primaryText,
              },
              '&:-webkit-autofill:focus': {
                '-webkitBoxShadow': `0 0 0px 1000px ${colors.inputBackground} inset`,
                '-webkit-text-fill-color': colors.primaryText,
              },
            },
            '&:hover': {
              border: '#282E36',
            },
            'input::placeholder': {
              color: colors.placeholderText,
            },
          },
          notchedOutline: {
            borderWidth: '1px !important',
            borderColor: `${additionalPalette?.colors?.border ?? colors.border} !important`,
            '.Mui-focused &': {
              borderColor: 'white !important',
            },
            '.MuiInputBase-colorWarning &': {
              borderColor: `${additionalPalette?.colors?.warningPrimary ?? colors.warningPrimary} !important`,
            },
            '.Mui-error &': {
              borderColor: `${additionalPalette?.colors?.errorPrimary ?? colors.errorPrimary} !important`,
            },
          },
        },
      },
      MuiInputLabel: {
        styleOverrides: {
          root: {
            'input ~ svg': {
              color: additionalPalette?.colors?.border ?? colors.border,
            },
            'input:checked ~ svg': {
              color: additionalPalette?.primary?.main ?? colors.primary,
            },
          },
        },
      },
      MuiButton: {
        styleOverrides: {
          // eslint-disable-next-line complexity
          root: ({ ownerState }: { ownerState: ButtonProps }) => ({
            textTransform: 'none',
            ...(ownerState.color === 'primary' && {
              color: colors.primaryText,
              '&:disabled': {
                backgroundColor: additionalPalette?.colors?.disabledColor ?? colors.disabledColor,
              },
            }),
            ...(ownerState.color === 'primary' &&
              ownerState.variant === 'contained' && {
                color: colors.primaryText,
                '&:hover': {
                  backgroundColor: additionalPalette?.colors?.hoverColor ?? colors.hoverColor,
                },
                '&:disabled': {
                  backgroundColor: additionalPalette?.colors?.disabledColor ?? colors.disabledColor,
                },
              }),
            ...(ownerState.color === 'secondary' && {
              color: colors.primaryText,
              borderColor: colors.secondaryText,
              backgroundColor: 'inherit',
              '&:hover': {
                borderColor: colors.secondaryText,
                backgroundColor: 'inherit',
              },
            }),
            ...(ownerState.color === 'secondary' &&
              ownerState.variant === 'contained' && {
                backgroundColor: colors.clickableGray,
                '&:hover': {
                  borderColor: colors.secondaryText,
                  backgroundColor: colors.clickableGrayHover,
                },
              }),
          }),
        },
      },
      MuiIconButton: {
        styleOverrides: {
          root: {
            backgroundColor: colors.clickableGray,
            '&.Mui-disabled': {
              backgroundColor: colors.clickableGray,
              opacity: '50%',
            },
          },
        },
      },
      MuiTooltip: {
        styleOverrides: {
          arrow: {
            color: colors.secondaryText,
          },
          tooltip: {
            backgroundColor: colors.secondaryText,
            padding: '4px',
            fontSize: 10,
            borderRadius: 3,
            fontWeight: 400,
          },
          tooltipPlacementTop: {
            marginBottom: '4px !important', // Hard to override tooltipPlacementTop
          },
        },
        defaultProps: {
          placement: 'top',
        },
      },
      MuiBottomNavigationAction: {
        styleOverrides: {
          root: {
            color: colors.secondaryText,
            '&.Mui-selected': {
              color: colors.primaryText,
              '.MuiTouchRipple-root': {
                height: '2px',
                width: '48px',
                marginLeft: 'auto',
                marginRight: 'auto',
                backgroundColor: additionalPalette?.primary?.main ?? colors.primary,
              },
            },
          },
        },
      },
      MuiTabs: {
        styleOverrides: {
          root: {
            '.MuiTabs-indicator': {
              height: '1px',
            },
          },
        },
      },
      MuiCardActionArea: {
        styleOverrides: {
          root: {
            ':hover': {
              backgroundColor: colors.clickableGrayHover,
            },
          },
        },
      },
      MuiFormHelperText: {
        styleOverrides: {
          root: {
            fontSize: '10px',
            '.MuiInputBase-colorWarning ~ &': {
              color: `${additionalPalette?.colors?.warningPrimary ?? colors.warningPrimary} !important`,
            },
          },
        },
      },
      MuiList: {
        styleOverrides: {
          root: {
            backgroundColor: colors.appBackground,
            padding: 8,
          },
        },
      },
      MuiMenuItem: {
        styleOverrides: {
          root: {
            height: '23px',
            minHeight: '23px',
            padding: '4px 8px',
            borderRadius: '3px',
            fontSize: 12,
            backgroundColor: colors.appBackground,
            '&:hover': {
              backgroundColor: colors.clickableGrayHover,
            },
            '&.Mui-selected': {
              backgroundColor: colors.nonClickableGray,
              color: colors.placeholderText,
              '&:hover': {
                backgroundColor: colors.nonClickableGray,
              },
            },
          },
        },
      },
    },
  });

export default defaultTheme;
