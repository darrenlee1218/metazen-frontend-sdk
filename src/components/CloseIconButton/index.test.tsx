import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/material/styles';

import CloseIconButton from './index';
import theme from '@theme/theme';

const onClick = jest.fn();

describe('Test snapshot CloseIconButton components', () => {
  it('should be render correctly', () => {
    const tree = TestRenderer.create(
      <ThemeProvider theme={theme()}>
        <CloseIconButton onClick={onClick} />
      </ThemeProvider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
