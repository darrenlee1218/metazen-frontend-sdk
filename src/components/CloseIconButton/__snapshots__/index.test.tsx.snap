// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot CloseIconButton components should be render correctly 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-14t6pro-MuiButtonBase-root-MuiIconButton-root"
  data-testid="close-icon-button"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-13obobm-MuiSvgIcon-root"
    data-testid="CloseIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</button>
`;
